Precisely preserve the timestamps when stripping debugging symbols

livecodestephen opened the pull request:
The `--preserve-dates` flag for `strip` and `objcopy` only has whole
second resolution.  This meant that the modification time was being
truncated to an integer number of seconds.  Consequently, when
stripping a file, some of the dependencies of that file would appear
to be newer, and hence the file would be unnecessarily remade during a
second call to make.

Simple solution: copy the timestamps to a separate file before we
strip and copy them back after.

livecodestephen committed:
Precisely preserve the timestamps when stripping debugging symbols

The `--preserve-dates` flag for `strip` and `objcopy` only has whole
second resolution.  This meant that the modification time was being
truncated to an integer number of seconds.  Consequently, when
stripping a file, some of the dependencies of that file would appear
to be newer, and hence the file would be unnecessarily remade during a
second call to make.

Simple soultion: copy the timestamps to a separate file before we
strip and copy them back after.

livecodestephen committed:
Remove the correct file in tools/extract-debug-symbols.sh

livecodeali commented:
@livecode-vulcan review ok 005d6d6

livecode-vulcan commented:
💙 review by @livecodeali ok 005d6d6ed5bc75229e451ee3f57eb3378d84404e

livecode-vulcan commented:
😞 test failure 005d6d6ed5bc75229e451ee3f57eb3378d84404e
- try-linux-x86: [failure](https://vulcan.livecode.com/builders/try-linux-x86/builds/725)
- try-win-x86: [pending](https://vulcan.livecode.com/builders/try-win-x86/builds/785)
- try-ios: [pending](https://vulcan.livecode.com/builders/try-ios/builds/675)
- try-android-armv6: [success](https://vulcan.livecode.com/builders/try-android-armv6/builds/739)
- try-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/670)
- try-mac: [success](https://vulcan.livecode.com/builders/try-mac/builds/662)

runrevmark commented:
It would seem that the --attributes-only option is understood on vulcan :(

livecodestephen commented:
The `--attributes-only` option was added in [coreutils 8.6](http://savannah.gnu.org/forum/forum.php?forum_id=6553) and works on my laptop which has 8.21.  Which version is vulcan using?

livecodefraser commented:
https://packages.debian.org/squeeze/coreutils

livecodestephen committed:
Ignore failure to copy attributes in extract-debug-symbols.sh

The 32-bit linux build is built on Debian Squeeze which only has coreutils 8.5.  Version >= 8.6 is required for the `--attributes-only` option on `cp`, so ignore failure of these commands.

livecodestephen commented:
I've changed it so that any failure of the cp command are ignored.

livecode-vulcan commented:
📝 Hi @livecodestephen, I haven't been able to verify that you've signed our [LiveCode Contributor's Agreement](https://livecode.com/account/developer/contribute).

If you have previously signed the Contributor's Agreement, I may not be able to detect it because you haven't [linked your GitHub account to your LiveCode account](https://livecode.com/account/developer/github).

Please see the [information for contributors](https://github.com/runrev/livecode/blob/develop/HACKING.md) for more information.

livecodefraser commented:
@livecode-vulcan review okay b7180c7

livecode-vulcan commented:
💙 review by @livecodefraser ok b7180c724aa8e470cf601922c0ce48e9d8920eaf

livecodesebastien commented:
@livecodestephen: Could you sign the CLA, so that we can merge that it? Thanks

livecodestephen commented:
I didn't realise that was required for LiveCode staff.  Will do it now.

livecode-vulcan commented:
😎 test success b7180c724aa8e470cf601922c0ce48e9d8920eaf
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/313)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/313)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/312)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/311)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/316)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/308)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/319)

livecodeali commented:
I have reverted this as it was causing problems with the build - running under gdb I got the error message

`"Workspace/livecode/linux-x86_64-bin/LiveCode-Community": not in executable format: File truncated`

livecodestephen commented:
Which system was having problems?  I'll see if I can reproduce it.

livecodeali commented:
I can reproduce on my Debian Wheezy VM
